### ISSUE TITLE
set DOCKER_CONTENT_TRUST variable when pushing a docker image to a cf application.

### ISSUE THREAD

**Issue by dagda1** [cloudfoundry/cli#1490]
**Please provide details on the following items. Failure to do so may result in deletion of your feature request.** 

There is currently no way to set `DOCKER_CONTENT_TRUST` while the process that does the docker pull can check whether the image has been signed.

If I try and pull an image locally that is not signed with the var set:

`export DOCKER_CONTENT_TRUST=1`

And I do a docker pull on an unsigned image, I get:

`docker pull xx/yyy:v0.3`

I get the following error:

`No valid trust data for v0.3`

But if I do the following 

```
CF_DOCKER_PASSWORD=xxx cf push --health-check-type process xxx --docker-image xxx/yyy:v0.3 -f ./manifest.yml --docker-username fff
```

The image installs ok because I presume the var has not been set.

**What's the user value of this feature request?**
The value is we can check the integrity of the images.

**Who is the functionality for?**
Anyone using docker containers.

**How often will this functionality be used by the user?**
Whenever pushing with a docker container.

**Who else is affected by the change?**
I don't think so.

**Is your feature request related to a problem? Please describe.**
Not that I can find

**Describe the solution you'd like**
The ability to set the `DOCKER_CONTENT_TRUST` variable

**Describe alternatives you've considered**
I have tried setting the var in the manifest.yml file but no cigar.
Tried `cf set-env` and a restage.
Tried `--no-start` and cf start

**Comment by tjvman** [comment]
@dagda1 In order to support this functionality, Diego would need to accept this environment variable when it's pulling down the docker image, and the Cloud Controller would need to add a field or endpoint that allows setting this environment variable and corresponding value. 

You should open an issue against [cloudfoundry/dockerapplifecycle](https://github.com/cloudfoundry/dockerapplifecycle), which is the component that handles pulling down the docker image to see if they're interested in supporting this workflow. Alternatively, you can reach the Diego maintainers on slack in the `#diego` channel in cloudfoundry.slack.com.

**Comment by tjvman** [comment]
Closing this issue, but feel free to reopen (or make a new issue) once the server-side is in place.
